### PR TITLE
Resolve Bug 558: InvalidAccessTagCheck needs to ignore ways that cross borders.

### PR DESCRIPTION
**Pull request by mselaineleong**
### Description:

Resolve Bug 558: InvalidAccessTagCheck needs to ignore ways that cross boarders.
https://github.com/osmlab/atlas-checks/issues/558

### Test Results:

Tested with 8 countries: ISR, CHN, PER, CYP, ISL, LUX, MLT, NZL

The additional boundary check is successful in removing the following flags:
ISR:
way 392663748
way 945594152
way 207971694
way 392653795
way 209209836
way 30889983
way 252722032
way 423697615

CHN:
way 252128471

CYP:
way 203114788
way 855617423
way 891673822

PER:
way 273953938

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java [hunk 1]
```diff
@@ -121,6 +106,22 @@ private boolean hasBoundaryNode(final AtlasObject object)
...
+    /**
+     * Checks if {@link AtlasObject} contains synthetic boundary Node
+     *
+     * @param object
+     *            the {@link AtlasObject} to check
+     * @return true if edge contains synthetic boundary Node.
+     */
+    private boolean hasBoundaryNode(final AtlasObject object)
+    {
+        if (object instanceof Edge && ((Edge) object).connectedNodes().stream().anyMatch(SyntheticBoundaryNodeTag::isBoundaryNode))
```
Comment (atiannicelli) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java:118]: this his be changed to a return (and get rid of the rest of the function)?
```suggestion
        return (object instanceof Edge && ((Edge) object).connectedNodes().stream().anyMatch(SyntheticBoundaryNodeTag::isBoundaryNode))
```

Comment (mselaineleong) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java]: Fixed! Thanks!

**Review by atiannicelli — APPROVED** [review]
LGTM

**Review by reichg — APPROVED** [review]
Nice addition! Looks good.

**Comment by reichg** [comment]
Only comment I have is to clean up the text in the PR. Example being the boilerplate false positive analysis with description of unit tests.

**Review by Bentleysb — APPROVED** [review]
Looks great, thanks!

**Review by sayas01 — APPROVED** [review]
LGTM!

**Pull request merged**